packages/calypso-e2e/src/jest-playwright-config/environment.ts

```diff
@@ -33,26 +44,74 @@ class JestEnvironmentPlaywright extends JestEnvironmentNode {
 	constructor( config: Config.ProjectConfig, context: Context ) {
 		super( config );
 
+		this.testFilePath = context.testPath;
 		this.testFilename = path.parse( context.testPath ).name;
 		this.testArtifactsPath = '';
 	}
 
+	/**
+	 * Determine the browser to be used for the spec.
+	 *
+	 * By default, the browser used is defined in the BROWSER_NAME
+	 * environment variable, but at times it may be required to
+	 * run tests in a different browser.
+	 *
+	 * To specify a different browser than default,
+	 * use the @browser tag in the docblock.
+	 *
+	 * Example:
+	 *
+	 * 	`@browser firefox`
+	 */
+	async determineBrowser(): Promise< BrowserType > {
```

Comment:
This piece is the change that allows the framework to switch the browser to be used for tests, taking into account:
- the `@browser` tag in the docblock, if present;
- the `BROWSER_NAME` environment variable, if present;